### PR TITLE
Authlist implementation

### PR DESCRIPTION
**Pull request by sebastian-aranda**
This PR has all the Frontend implementation related to the Authlist component.

### CODE REVIEW

File: love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+      selectedCSC,
+      selectedUser,
+      keywords,
+      cscOptions,
+      userOptions,
+      removeIdentityModalShown,
+      removeIdentityModalText,
+    } = this.state;
+
+    // console.log(authlistAdminPermission);
```
Comment (areyesd14) [inline love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx:392]: This console.log must be removed

File: love/src/redux/selectors/selectors.js [hunk 2]
```diff
@@ -425,6 +427,20 @@ export const getLATISSState = (state) => {
   };
 };
 
+export const getAuthlistState = (state, subscriptions) => {
+  // const authlistData = getStreamsData(state, subscriptions);
+  // console.log(authlistData);
```
Comment (areyesd14) [inline love/src/redux/selectors/selectors.js:432]: If this code is not used, it should be removed

File: love/src/redux/selectors/selectors.js [hunk 3]
```diff
@@ -425,6 +427,20 @@ export const getLATISSState = (state) => {
...
+  const authlistData = {
```
Comment (areyesd14) [inline love/src/redux/selectors/selectors.js:433]: This data must come from the backend

File: love/src/redux/selectors/selectors.js [hunk 4]
```diff
@@ -425,6 +427,20 @@ export const getLATISSState = (state) => {
...
+    'event-ATDome-0-authList': { authorizedUsers: 'tribeiro@nb-tribeiro', nonAuthorizedCSCs: 'MTPtg:1' },
```
Comment (areyesd14) [inline love/src/redux/selectors/selectors.js:434]: This information must come from the backend

File: love/src/redux/selectors/selectors.js [hunk 5]
```diff
@@ -425,6 +427,20 @@ export const getLATISSState = (state) => {
...
+    'event-ScriptQueue-1-authList': { authorizedUsers: 'tribeiro@nb-tribeiro', nonAuthorizedCSCs: 'MTPtg:1' },
+    'event-ATMCS-0-authList': {
+      authorizedUsers: 'saranda@inria-ThinkPad-P50-3,tribeiro@nb-tribeiro',
```
Comment (areyesd14) [inline love/src/redux/selectors/selectors.js:437]: Same as in the previous comment

File: love/src/Utils.js [hunk 6]
```diff
@@ -388,6 +389,142 @@ export default class ManagerInterface {
...
+      // console.log('Token not found during validation');
+      return new Promise((resolve) => resolve(false));
+    }
+    const url = `${this.getApiBaseUrl()}authlistrequest/`;
+    return fetch(url, {
+      method: 'GET',
+      headers: this.getHeaders(),
+    }).then((response) => {
+      if (response.status >= 500) {
+        // console.error('Error communicating with the server.);
```
Comment (areyesd14) [inline love/src/Utils.js:431]: This console.log must be removed or replaced by a toast element

File: love/src/Utils.js [hunk 7]
```diff
@@ -388,6 +389,142 @@ export default class ManagerInterface {
...
+        return false;
+      }
+      if (response.status === 401 || response.status === 403) {
+        // console.log('Session expired. Logging out');
```
Comment (areyesd14) [inline love/src/Utils.js:435]: This console.log must be removed or replaced by a toast element

File: love/src/Utils.js [hunk 8]
```diff
@@ -388,6 +389,142 @@ export default class ManagerInterface {
...
+    username,
+    cscsToChange,
+    authorizedUsers,
+    nonAuthorizedCSCs,
+    message = null,
+    duration = null,
+  ) {
+    const token = ManagerInterface.getToken();
+    if (token === null) {
+      // console.log('Token not found during validation');
```
Comment (areyesd14) [inline love/src/Utils.js:455]: This console.log must be removed or replaced by a toast element

File: love/src/components/AuthList/AdminAuthList/AdminAuthList.jsx [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+      moreModalShown,
+      message,
+      duration,
+      selectedStartDate,
+      selectedEndDate,
+      selectedRequest,
+      authListRequests,
+    } = this.state;
+
+    // console.log(authListRequests);
```
Comment (areyesd14) [inline love/src/components/AuthList/AdminAuthList/AdminAuthList.jsx:289]: This console.log must be removed

File: love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx [hunk 10]
```diff
@@ -0,0 +1,483 @@
...
+      const cscsToChange = selectedCSC;
+      const authorizedUsers = authlist.authorizedUsers
+        .split(',')
+        .map((x) => `-${x}`)
+        .join(',');
+      const nonAuthorizedCSCs = authlist.nonAuthorizedCSCs
+        .split(',')
+        .map((x) => `-${x}`)
+        .join(',');
+      console.log('Request: ', cscsToChange, authorizedUsers, nonAuthorizedCSCs);
```
Comment (areyesd14) [inline love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx:193]: Is this console.log must be replaced by the next line?

File: love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx (hunk 1 above)
Comment (sebastian-aranda) [reply love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx]: I've removed this comment on further commits. Thanks!

File: love/src/redux/selectors/selectors.js (hunk 2 above)
Comment (sebastian-aranda) [reply love/src/redux/selectors/selectors.js]: I've removed this comment on further commits. Thanks!

File: love/src/redux/selectors/selectors.js (hunk 4 above)
Comment (sebastian-aranda) [reply love/src/redux/selectors/selectors.js]: I've refactored this on further commits. Thanks!

File: love/src/redux/selectors/selectors.js (hunk 5 above)
Comment (sebastian-aranda) [reply love/src/redux/selectors/selectors.js]: Removed.

File: love/src/Utils.js [hunk 11]
```diff
@@ -388,6 +389,142 @@ export default class ManagerInterface {
...
+      return response.json().then((resp) => {
+        return resp;
+      });
+    });
+  }
+
+  static getAuthListRequests() {
+    const token = ManagerInterface.getToken();
+    if (token === null) {
+      // console.log('Token not found during validation');
```
[the remark this answers is not in the document]
Comment (sebastian-aranda) [reply love/src/Utils.js]: I've removed this comment on further commits. Thanks!

File: love/src/Utils.js (hunk 6 above)
Comment (sebastian-aranda) [reply love/src/Utils.js]: I've removed this comment on further commits. Thanks!

File: love/src/Utils.js (hunk 7 above)
Comment (sebastian-aranda) [reply love/src/Utils.js]: I've removed this comment on further commits. Thanks!

File: love/src/Utils.js (hunk 8 above)
Comment (sebastian-aranda) [reply love/src/Utils.js]: I've removed this comment on further commits. Thanks!

File: love/src/components/AuthList/AdminAuthList/AdminAuthList.jsx (hunk 9 above)
Comment (sebastian-aranda) [reply love/src/components/AuthList/AdminAuthList/AdminAuthList.jsx:302]: I've removed this comment on further commits. Thanks!

File: love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx (hunk 10 above)
Comment (sebastian-aranda) [reply love/src/components/AuthList/SummaryAuthList/SummaryAuthList.jsx]: I've removed this on further commits. Thanks!

**Review by areyesd14 — APPROVED** [review]
Great job, Sebastián! Thanks a lot!

**Pull request merged**